tests/integration-async/migrator/test_migrator_api.py

```diff
@@ -0,0 +1,329 @@
+
+import time
+import zlib
+
+import pytest
+from invenio_pidstore.errors import PIDDoesNotExistError
+
+from inspirehep.migrator.api import continuous_migration
+from inspirehep.records.api import InspireRecord
+from inspirehep.search.api import InspireSearch
+
+
+def test_continuous_migration(
+    app, cache, celery_app_with_context, celery_session_worker
+):
+    raw_record_citer = (
+        b"<record>"
+        b'  <controlfield tag="001">666</controlfield>'
+        b'  <datafield tag="245" ind1=" " ind2=" ">'
+        b'    <subfield code="a">This is a citer record</subfield>'
+        b"  </datafield>"
+        b'  <datafield tag="980" ind1=" " ind2=" ">'
+        b'    <subfield code="a">HEP</subfield>'
+        b"  </datafield>"
+        b'   <datafield tag="999" ind1="C" ind2="5">'
+        b'    <subfield code="0">667</subfield>'
+        b'    <subfield code="h">Achasov, M.N.</subfield>'
+        b'    <subfield code="k">snd-2018</subfield>'
+        b'    <subfield code="m">(SND Collaboration)</subfield>'
+        b'    <subfield code="o">2</subfield>'
+        b'    <subfield code="s">Phys.Rev.,D97,012008</subfield>'
+        b'    <subfield code="x">'
+        b"    [2] M. N. Achasov (SND Collaboration), Phys. Rev. D 97, 012008 (2018)."
+        b"    </subfield>"
+        b'    <subfield code="y">2018</subfield>'
+        b'    <subfield code="z">0</subfield>'
+        b'    <subfield code="z">1</subfield>'
+        b"    </datafield>"
+        b"</record>"
+    )
+    citer_control_number = 666
+
+    raw_record_citing = (
+        b"<record>"
+        b'  <controlfield tag="001">667</controlfield>'
+        b'  <datafield tag="245" ind1=" " ind2=" ">'
+        b'    <subfield code="a">This is a citing record</subfield>'
+        b"  </datafield>"
+        b'  <datafield tag="980" ind1=" " ind2=" ">'
+        b'    <subfield code="a">HEP</subfield>'
+        b"  </datafield>"
+        b"</record>"
+    )
+    citing_control_number = 667
+
+    cache.rpush("legacy_records", zlib.compress(raw_record_citer))
+    cache.rpush("legacy_records", zlib.compress(raw_record_citing))
+
+    assert cache.llen("legacy_records") == 2
+
+    continuous_migration()
+
+    # I don't like timeouts, it's the only way to wait for this chain
+    time.sleep(10)
+
+    record_citer = InspireRecord.get_record_by_pid_value(citer_control_number, "lit")
+    record_citing = InspireRecord.get_record_by_pid_value(citing_control_number, "lit")
+
+    assert record_citing.citation_count == 1
+
+    record_citer_es = InspireSearch.get_record_data_from_es(record_citer)
+    result_citer_control_number = record_citer_es["control_number"]
+
+    assert citer_control_number == result_citer_control_number
+
+    record_citing_es = InspireSearch.get_record_data_from_es(record_citing)
+    result_citing_control_number = record_citing_es["control_number"]
+
+    assert citing_control_number == result_citing_control_number
```

Comment:
[we could also check that the migrator indexed properly the record by checking the citation_count of ES, WDYT?

Reply:
good point